src/utils.ts

```diff
@@ -505,6 +527,155 @@ export function isNodeModule(path: string) {
   return /^([A-z0-9@])/.test(path)
 }
 
+type NexusWrapKind = { type: 'List'; nullable: boolean } | { type: 'WrappedType'; nullable: boolean }
+
+export function unwrapNexusDef(
+  typeDef: AllNexusTypeDefs | AllNexusArgsDefs | string,
+  nonNullDefault: boolean
+): { namedType: AllNexusNamedTypeDefs | AllNexusArgsDefs | string; wrapping: NexusWrapKind[] } {
+  const wrapping: NexusWrapKind[] = []
+  let namedType = typeDef
+
+  const isWrappedType = (obj: any): obj is AllNexusNamedTypeDefs | NexusArgDef<any> | string =>
+    isNexusNamedTypeDef(obj) || isNexusArgDef(obj) || typeof obj === 'string'
+
+  if (isWrappedType(typeDef)) {
+    return { namedType: typeDef, wrapping: [{ type: 'WrappedType', nullable: !nonNullDefault }] }
+  }
+
+  while (isNexusWrappingType(namedType)) {
+    // nullable(list(Type))
+    if (isNexusNullTypeDef(namedType) && isNexusListTypeDef(namedType.ofType)) {
+      wrapping.unshift({ type: 'List', nullable: true })
+      namedType = namedType.ofType.ofType
+    }
+
+    // nullable(Type)
+    if (isNexusNullTypeDef(namedType) && isWrappedType(namedType.ofType)) {
+      wrapping.unshift({ type: 'WrappedType', nullable: true })
+      namedType = namedType.ofType
+      break
+    }
+
+    // nonNull(list(Type))
+    if (isNexusNonNullTypeDef(namedType) && isNexusListTypeDef(namedType.ofType)) {
+      wrapping.unshift({ type: 'List', nullable: false })
+      namedType = namedType.ofType.ofType
+    }
+
+    // nonNull(Type)
+    if (isNexusNonNullTypeDef(namedType) && isWrappedType(namedType.ofType)) {
+      wrapping.unshift({ type: 'WrappedType', nullable: false })
+      namedType = namedType.ofType
+      break
+    }
+
+    // list(...)
+    if (isNexusListTypeDef(namedType)) {
+      wrapping.unshift({ type: 'List', nullable: !nonNullDefault })
+      namedType = namedType.ofType
+    }
+
+    // Type
+    if (!isNexusWrappingType(namedType)) {
+      wrapping.unshift({ type: 'WrappedType', nullable: !nonNullDefault })
+    }
+  }
+
+  return { namedType, wrapping }
+}
+
+export function wrapType(
```

Comment:
```suggestion
/**
 * Take a type and adjust the wrapping to correctly reflect the nullability defaults. For example a list(Foo) with nullable by default would become nullable(list(Foo)).
 */
export function rewrapTypeWithNullabilityDefaults(
```

Reply:
> Take a type and adjust the wrapping to correctly reflect the nullability defaults. For example a list(Foo) with nullable by default would become nullable(list(Foo))

This function is not only there to reflect the nullability defaults. It's mainly here to convert nexus wrappers to GraphQL wrappers (and while doing that, we also reflect the nullability defaults indeed)

Reply:
I suggest renaming it to `wrapAsGraphQLType`, which mirrors the existing `wrapAsNexusType`

Reply:
or `rewrapAsGraphQLType`

Reply:
named to `rewrapAsGraphQLType`